core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -123,7 +123,7 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
    * @see [[handleError]] to map to an `A` value instead of `Unit`.
    * @see [[https://github.com/typelevel/cats-effect/issues/3152 cats-effect#3152]]
    */
-  def voidError(fu: F[Unit]): F[Unit] = handleError(fu)(Function.const(()))
+  def voidError(fu: F[_]): F[Unit] = handleError(fu)(Function.const(()))
```

Comment:
I'm not sure this is correct. You can get runtime cast errors. See a similar situation in:
- https://github.com/typelevel/cats/pull/4352#discussion_r1027255402

Reply:
Btw, there was a discussion about it in the initial PR:
https://github.com/typelevel/cats/pull/4324#discussion_r1006331939